### DISCUSSION TITLE
Can't get dependencies to upload to npm

### DISCUSSION THREAD

**Discussion by othenos** [shannah/jdeploy#122] [General]
I've been using update4j for several years but am intrigued by the simpler mechanism jDeploy seems to offer. However, I cannot figure out why my dependencies are not getting upload when publishing to npm.

I generate a runnable jar in the target/app directory, along with a folder will all the dependencies 
![Screenshot 2024-05-05 at 3 40 01 PM](https://github.com/shannah/jdeploy/assets/1975212/636258bd-80cb-4135-a730-2e7f780e1ae2)

When I publish to npm the jar goes but not the dependencies folder. I'm at a bit of a loss as to how the folder and contents would get uploaded in the first place, since does not seem to be a place in the package.json to describe them. That is unless it is required that they be entered manually in the file. That would be a bit daunting as there are dozens of dependencies.

If I missed somethings in the documentation (which is probably the case), could you point me to the proper  place where I can understand this part of the process?

I'm looking forward to getting this working. Thanks for your efforts

**Comment by othenos** [comment]
Looks like a good night of sleep was all I needed. It looks like I just did not create the jar properly. It now published as expected.

**Comment by shannah** [comment]
Thanks, Mike for posting both the question and the answer.  Sorry I didn't manage to reply before you solved it yourself.  Just been a little swamped this week and fell behind.

Let me know if you have any more questions, or feedback.

Steve
